packages/core/src/utils.ts

```diff
@@ -988,24 +947,32 @@ export const getSphinxWalletPrivateKey = (walletIndex: number): string => {
  */
 export const addSphinxWalletsToGnosisSafeOwners = async (
   safeAddress: string,
+  moduleAddress: string,
   executionMode: ExecutionMode,
   provider: SphinxJsonRpcProvider | HardhatEthersProvider
 ): Promise<Array<ethers.Wallet>> => {
-  // The caller of the transactions on the Gnosis Safe will be the Gnosis Safe itself. This is
-  // necessary to prevent the calls from reverting.
-  const safeSigner = await getImpersonatedSigner(safeAddress, provider)
+  // The caller of the transactions on the Gnosis Safe will be the Sphinx Module. This is necessary
+  // to prevent the calls from reverting. An alternative approach is to call the Gnosis Safe from
+  // the Gnosis Safe itself. However, this would increment its nonce in the `addOwnerWithThreshold`
+  // calls that occur later in this function, which would mess up the addresses of contracts
+  // deployed via the Gnosis Safe. We can't reset the Gnosis Safe's nonce via `hardhat_setNonce`
+  // because Hardhat throws an error if we attempt to set a contract's nonce lower than its current
+  // nonce.
```

Comment:
Had to change the implementation of this function and `removeSphinxWalletsFromGnosisSafeOwners` to avoid this edge case